src/iap/shared/validators/match-password.validator.ts

```diff
@@ -0,0 +1,20 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+
+export function matchPassword(password: string, passwordConfirmation: string): ValidatorFn {
+	return (control: AbstractControl): ValidationErrors | null => {
+		const passwordControl = control.get(password);
+		const passwordConfirmationControl = control.get(passwordConfirmation);
+
+		if (!passwordConfirmationControl || !passwordControl) {
+			return null;
+		}
+
+		if (passwordControl.value !== passwordConfirmationControl.value) {
+			passwordConfirmationControl.setErrors({ passwordMismatch: true });
+			return { passwordMismatch: true };
```

Comment:
:point_right: :point_left: 
```suggestion
			passwordConfirmationControl.setErrors({ passwordMismatch: true });
			
			return { passwordMismatch: true };
```